provider/pkg/provider/provider.go

```diff
@@ -1528,9 +1528,19 @@ func (k *kubeProvider) Create(
 	}
 	initialized, awaitErr := await.Creation(config)
 	if awaitErr != nil {
-		if req.GetPreview() && k.isDryRunDisabledError(err) {
-			logger.V(9).Infof("could not preview Create(%v): %v", urn, err)
-			return &pulumirpc.CreateResponse{Id: "", Properties: req.GetProperties()}, nil
+		if req.GetPreview() {
+			failedPreview := func() bool {
+				_, isPreviewErr := awaitErr.(await.PreviewError)
```

Comment:
Could you please explain how `&namespaceError{c.Inputs}` becomes `PreviewError`?

Reply:
I think I was a bit tired on that day. I understand it now.

Reply:
`PreviewError` is an interface, and `namespaceError` implements that interface by defining the `Object` pointer method. I'm not sure this approach is the simplest option, but I'm copying the existing approach taken by this provider.